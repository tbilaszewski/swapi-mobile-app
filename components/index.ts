export * from "./PeopleCard/";
export * from "./SpaceshipCard/";
export * from "./CardNotFound";
export * from "./ScoreDisplay";
