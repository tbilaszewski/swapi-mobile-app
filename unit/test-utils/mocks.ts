import { PeopleResponse, SpaceshipResponse } from "../../repository/model";

export const personDataMock: PeopleResponse = {
  name: "Luke Skywalker",
  height: "172",
  mass: "77",
  hair_color: "blond",
  skin_color: "fair",
  eye_color: "blue",
  birth_year: "19BBY",
  gender: "male",
  homeworld: "https://swapi.dev/api/planets/1/",
  films: [
    "https://swapi.dev/api/films/2/",
    "https://swapi.dev/api/films/6/",
    "https://swapi.dev/api/films/3/",
    "https://swapi.dev/api/films/1/",
    "https://swapi.dev/api/films/7/",
  ],
  species: ["https://swapi.dev/api/species/1/"],
  vehicles: [
    "https://swapi.dev/api/vehicles/14/",
    "https://swapi.dev/api/vehicles/30/",
  ],
  starships: [
    "https://swapi.dev/api/starships/12/",
    "https://swapi.dev/api/starships/22/",
  ],
  created: "2014-12-09T13:50:51.644000Z",
  edited: "2014-12-20T21:17:56.891000Z",
  url: "https://swapi.dev/api/people/1/",
};

export const spaceshipDataMock: SpaceshipResponse = {
  name: "Death Star",
  model: "DS-1 Orbital Battle Station",
  manufacturer:
    "Imperial Department of Military Research, Sienar Fleet Systems",
  cost_in_credits: "1000000000000",
  length: "120000",
  max_atmosphering_speed: "n/a",
  crew: "342,953",
  passengers: "843,342",
  cargo_capacity: "1000000000000",
  consumables: "3 years",
  hyperdrive_rating: "4.0",
  MGLT: "10",
  starship_class: "Deep Space Mobile Battlestation",
  pilots: [],
  films: ["http://swapi.dev/api/films/1/"],
  created: "2014-12-10T16:36:50.509000Z",
  edited: "2014-12-20T21:26:24.783000Z",
  url: "http://swapi.dev/api/starships/9/",
};
