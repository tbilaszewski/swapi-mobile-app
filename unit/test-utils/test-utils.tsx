import { render } from "@testing-library/react-native";
import { Provider } from "react-redux";
import React from "react";
import { store } from "../../store";

const AllTheProviders: React.FC = ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};

const customRender = (el: React.ReactElement, options = {}) =>
  render(el, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react-native";

export { customRender as render };
