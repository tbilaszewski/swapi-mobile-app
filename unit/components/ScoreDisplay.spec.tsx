import React from "react";
import { ScoreDisplay } from "../../components";
import { render } from "../test-utils";

describe("ScoreDisplay", () => {
  it("should match snapshot ", () => {
    const cardWrapper = render(<ScoreDisplay />);
    expect(cardWrapper.toJSON()).toMatchSnapshot();
  });

  it("should have both score with initial values ", () => {
    const { getAllByText } = render(<ScoreDisplay />);
    expect(getAllByText("0").length).toBe(2);
  });
});
