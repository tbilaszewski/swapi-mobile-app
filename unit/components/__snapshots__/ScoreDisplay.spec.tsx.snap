// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreDisplay should match snapshot  1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "height": 100,
      "justifyContent": "space-around",
      "marginTop": 40,
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#6200ee",
            "borderRadius": 24,
            "height": 48,
            "width": 48,
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {},
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "textAlign": "center",
                "textAlignVertical": "center",
              },
              Object {
                "color": "#ffffff",
                "fontSize": 24,
                "lineHeight": 48,
              },
              undefined,
            ],
          ]
        }
      >
        0
      </Text>
    </View>
    <Text>
      Player 1
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "gray",
        "height": 50,
        "width": 2,
      }
    }
  />
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#6200ee",
            "borderRadius": 24,
            "height": 48,
            "width": 48,
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {},
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "textAlign": "center",
                "textAlignVertical": "center",
              },
              Object {
                "color": "#ffffff",
                "fontSize": 24,
                "lineHeight": 48,
              },
              undefined,
            ],
          ]
        }
      >
        0
      </Text>
    </View>
    <Text>
      Player 2
    </Text>
  </View>
</View>
`;
